### PR TITLE
Add link command and associated methods to add new links and check for links

### PR DESCRIPTION
**Pull request by lavanyagarg112**
Add link command and associated methods to add new links and check for links

Associated with #72

### CODE REVIEW

File: src/main/java/seedu/address/model/AddressBook.java [hunk 1]
```diff
@@ -94,6 +95,38 @@ public void removePerson(Person key) {
...
+    }
+
+    /**
+     * Adds a link between the patient and caregiver
+     * @param patientNric
+     * @param caregiverNric
+     */
+    public void addLink(String patientNric, String caregiverNric) {
+        Person patient = persons.get(patientNric);
+        Person caregiver = persons.get(caregiverNric);
```
Comment (pradyuprasad) [inline src/main/java/seedu/address/model/AddressBook.java:126]: Does persons.get have any way of avoiding a nullpointer reference? I would suggest that you handle this (ideally by using an Optional?)

File: src/main/java/seedu/address/model/person/UniquePersonList.java [hunk 2]
```diff
@@ -97,6 +97,40 @@ public void setPersons(List<Person> persons) {
...
+
+    /**
+     * Adds a link between the given patient and caregiver.
+     * @param patient
+     * @param caregiver
+     */
+    public void addLink(Person patient, Person caregiver) {
+        patient.addCaregiver(caregiver);
+        caregiver.addPatient(patient);
+    }
```
Comment (pradyuprasad) [inline src/main/java/seedu/address/model/person/UniquePersonList.java:132]: A few points
1. Perhaps indicate the successful operation to the caller of this method by returning a boolean
2. Handle the case of either of the inputs being null
3. Also handle the case if either of the persons is not in the list (possibly by throwing an exception?)

**Review by pradyuprasad — CHANGES_REQUESTED** [review]
Main changes are 
1. ensuring input validation for the addLink commands in both classes
2. indicating that the method succeeded or failed

File: src/main/java/seedu/address/model/AddressBook.java (hunk 1 above)
Comment (lavanyagarg112) [reply src/main/java/seedu/address/model/AddressBook.java:126]: Will do this, thanks for pointing it out

**Review by pradyuprasad — COMMENTED** [review]
LGTM

**Pull request merged**